### ISSUE TITLE
"OverflowError: Python int too large to convert to C long"

### ISSUE THREAD

**Issue by palmerf** [tbumi/glacier-upload#27]
Successfully uploaded a video file to Glacier using glacier_upload.  The size of the
file was 7406281365 bytes. Initiated an archival retrieval command using:

    init_archive_retrieval -v MyVault -a ArchiveName -d description

which completed successfully and returned a job id.
After getting an SNS notification that the the job completed successfully, I tried:

    get_glacier_job_output --vault-name MyVault  --job-id JobId -f archive.tar.xz
  
I get:

Checking job status...
Job status: Succeeded
Retrieving job data...
Traceback (most recent call last):
  File "C:\Users\palmerf\AppData\Local\Programs\Python\Python38\Scripts\get_glacier_job_output-script.py", line 11, in <module>
    load_entry_point('glacier-upload==1.2', 'console_scripts', 'get_glacier_job_output')()
  File "c:\users\palmerf\appdata\local\programs\python\python38\lib\site-packages\click\core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "c:\users\palmerf\appdata\local\programs\python\python38\lib\site-packages\click\core.py", line 782, in main
    rv = self.invoke(ctx)
  File "c:\users\palmerf\appdata\local\programs\python\python38\lib\site-packages\click\core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "c:\users\palmerf\appdata\local\programs\python\python38\lib\site-packages\click\core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "c:\users\palmerf\appdata\local\programs\python\python38\lib\site-packages\glacier_upload\get_job_output.py", line 41, in get_job_output
    file.write(response['body'].read())
  File "c:\users\palmerf\appdata\local\programs\python\python38\lib\site-packages\botocore\response.py", line 78, in read
    chunk = self._raw_stream.read(amt)
  File "c:\users\palmerf\appdata\local\programs\python\python38\lib\site-packages\urllib3\response.py", line 515, in read
    data = self._fp.read() if not fp_closed else b""
  File "c:\users\palmerf\appdata\local\programs\python\python38\lib\http\client.py", line 467, in read
    s = self._safe_read(self.length)
  File "c:\users\palmerf\appdata\local\programs\python\python38\lib\http\client.py", line 608, in _safe_read
    data = self.fp.read(amt)
  File "c:\users\palmerf\appdata\local\programs\python\python38\lib\socket.py", line 669, in readinto
    return self._sock.recv_into(b)
  File "c:\users\palmerf\appdata\local\programs\python\python38\lib\ssl.py", line 1241, in recv_into
    return self.read(nbytes, buffer)
  File "c:\users\palmerf\appdata\local\programs\python\python38\lib\ssl.py", line 1099, in read
    return self._sslobj.read(len, buffer)
OverflowError: Python int too large to convert to C long

Clearly, the size of the file is greater than C's 2,147,483,647 long int definition. Is there a way around this?

**Comment by avinash-oza** [comment]
In this line which is mentioned:

C:\users\palmerf\appdata\local\programs\python\python38\lib\site-packages\glacier_upload\get_job_output.py", line 41, in get_job_output

Can you try changing
```python
file.write(response['body'].read())
```

```python
while 1:
    data = file_output.read(3.2E7) # 32 MB chunk at a time
    if not data:
        break

    file.write(data)
```

This should provide a batching like read from the read rather than downloading it all at once. Referred to this example here: https://stackoverflow.com/a/4971681

If this works, I can submit a pull request to fix it.

**Comment by palmerf** [comment]
This is what worked:

    with open(file_name, "xb") as file:
         while 1:
         data = response["body"].read(int(2**25)  # 32 MB chunk at a time
         if not data:
             break

        file.write(data)

This is not ideal - especially if the archive was created from a list of files. In that case "upload.py"
creates a tarfile. 

Debugging this program is either time-consuming - a 4 hour wait for a download - or expensive if you add provisioned capacity. I've decided to go with one of the Windows solutions.

**Comment by tbumi** [comment]
This should be solved in [version 2.0](https://pypi.org/project/glacier-upload/2.0/) with chunked downloading of archives.
